torchtext/experimental/transforms.py

```diff
@@ -174,6 +175,162 @@ def to_ivalue(self):
+            - text_unigram_15000
+            - text_unigram_25000
+            - text_unigram_50000
+            - text_bpe_15000
+            - text_bpe_25000
+            - text_bpe_50000
+            Otherwise, the file path to the user-provided sentencepiece model is required.
+
+    Examples:
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+
+    """
+    if spm_model in _pretrained_spm:
+        spm_model = download_from_url('https://pytorch.s3.amazonaws.com/models/text/pretrained_spm/{}.model'.format(spm_model))
+        return load_sp_model(spm_model)
+    else:
+        raise RuntimeError('The pretrained sentencepiece model is not valid')
+
+
+class SentencePieceTokenizer(nn.Module):
+    r"""Tokenizer based on a pretained sentencepiece model.
+
+    Args:
+       spm_model: the sentencepiece model instance
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import SentencePieceTokenizer
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+        >>> spm_tokenizer = SentencePieceTokenizer(sp_model)
+        >>> jit_spm_tokenizer = torch.jit.script(spm_tokenizer)
+    """
+
+    def __init__(self, spm_model):
+        super(SentencePieceTokenizer, self).__init__()
+        self.sp_model = spm_model
+
+    def forward(self, lines: List[str]) -> List[List[str]]:
+        r"""
+        Args:
+            lines: a list of the input strings
+
+        Examples:
+            >>> spm_tokenizer(['the pretrained sp model names'])
+            >>> [['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names']]
+        """
+
+        tokens: List[List[str]] = []
+        for line in lines:
+            tokens.append(self.sp_model.EncodeAsPieces(line))
+        return tokens
+
+    @torch.jit.export
+    def decode(self, tokens_list: List[List[str]]) -> List[str]:
+        r"""
+        Args:
+            tokens_list: the tokens list for decoder
+
+        Examples:
+            >>> spm_transform.decoder([['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names']])
+            >>> ['the pretrained sp model names']
+        """
+        string_list: List[str] = []
+        for tokens in tokens_list:
+            string_list.append(self.sp_model.DecodePieces(tokens))
+        return string_list
+
+
+class SentencePieceTransform(nn.Module):
+    r"""String to ids transform based on a pretained sentencepiece model
+
+    Args:
+       spm_model: the sentencepiece model instance
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import SentencePieceTransform
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+        >>> spm_transform = SentencePieceTransform(sp_model)
```

Comment:
I wouldn't encourage users to call into the initialization method directly, instead we should continue to use the factory function pattern, so that we can change the init signature and also add new features without introducing bc breaking changes.

Reply:
Instead of this, you could add a method to SentencePiece like
```
def to_tokenizer(self):
    return SentencePieceTransform(self)
```

Then you can have 
```
sp_model = pretrained_spm('text_unigram_25000')
spm_transform = sp_model.to_tokenizer()
```

Reply:
Or you can have a factory function like

```
def sentence_piece_transform(spm_model: SentencePiece):
    return SentencePieceTransform(spm_model)
```